app/lib/domains.server.ts

```diff
@@ -0,0 +1,146 @@
+import * as dns from './dns.server';
+import { RecordStatus } from '@prisma/client';
+import { prisma } from '~/db.server';
+import type { RecordType } from '@prisma/client';
+
+interface DnsRecord {
+  type: RecordType;
+  name: string;
+  value: string;
+}
+
+interface Record {
+  id?: number;
+  username?: string;
+  description?: string;
+  course?: string;
+  ports?: string;
+}
+
+export const createUserDomain = async (data: Record & DnsRecord) => {
+  const existingRecords = await prisma.record.findMany({
+    where: {
+      name: data.name,
+      type: data.type,
+      value: data.value,
+    },
+  });
+
+  if (existingRecords.length > 0) {
+    throw new Error('Record already exists');
+  }
+
+  const changeId = await dns.createRecord(data.type, data.name, data.value);
```

Comment:
You don't seem to be using this value, so maybe don't bother doing it here?

Reply:
oh, I should remove them.